qtclient/requestmicpermission.mm

```diff
@@ -0,0 +1,33 @@
+#import "requestmicpermission.h"
+#import "AVFoundation/AVFoundation.h"
+
+bool RequestMicPermission::openMic()
+{
+    if (@available(macOS 10.14, *)) {
+
+        AVAuthorizationStatus st = [AVCaptureDevice authorizationStatusForMediaType:AVMediaTypeAudio];
+        if (st == AVAuthorizationStatusAuthorized) {
+          return true;
+        }
+
+        dispatch_group_t group = dispatch_group_create();
+
+        __block bool accessGranted = false;
+
+        if (st != AVAuthorizationStatusAuthorized) {
+          dispatch_group_enter(group);
+          [AVCaptureDevice requestAccessForMediaType:AVMediaTypeAudio completionHandler:^(BOOL granted) {
+
+              accessGranted = granted;
+              NSLog(@"Granted!");
+              dispatch_group_leave(group);
+            }];
+        }
+
+        dispatch_group_wait(group, dispatch_time(DISPATCH_TIME_NOW, (int64_t)(5.0 * NSEC_PER_SEC)));
```

Comment:
Was there a reason for choosing a 5 second timeout instead of DISPATCH_TIME_FOREVER?
https://developer.apple.com/documentation/dispatch/1452794-dispatch_group_wait

I imagine the user will be confused if they take longer but jammr times out before they closed the permission dialog. That's why an infinite timeout seems appropriate here. What do you think?